sklearn/compose/_column_transformer.py

```diff
@@ -453,8 +453,9 @@ def _get_feature_name_out_for_transformer(
                 f"Transformer {name} (type {type(trans).__name__}) does "
                 "not provide get_feature_names_out."
             )
-        if isinstance(column, Iterable) and not all(
-            isinstance(col, str) for col in column
+        if isinstance(column, slice) or (
+            isinstance(column, Iterable)
+            and not all(isinstance(col, str) for col in column)
```

Comment:
```suggestion
        if isinstance(column, (Iterable, slice))
            and not all(isinstance(col, str) for col in column
```

Reply:
@adrinjalali The suggested change does not implement the original logic. If column is a slice, then it is not Iterable and the code will fail.

Reply:
@adrinjalali Do you have any other suggestions to improve the commit? Tx